lib/breaker_box.ex

```diff
@@ -65,61 +65,72 @@ defmodule BreakerBox do
   anything you want. Re-using a breaker name in multiple places will overwrite
   with the last configuration.
   """
-  @spec register(breaker_name :: term, breaker_options :: BreakerConfiguration.t()) ::
+  @spec register(
+          breaker_name :: term,
+          breaker_options :: BreakerConfiguration.t(),
+          process_name :: atom()
+        ) ::
           :ok | :reset | {:error, reason :: term}
-  def register(breaker_name, %BreakerConfiguration{} = breaker_options) do
-    GenServer.call(__MODULE__, {:register, breaker_name, breaker_options})
+  def register(
+        breaker_name,
+        %BreakerConfiguration{} = breaker_options,
+        process_name \\ __MODULE__
+      ) do
+    GenServer.call(process_name, {:register, breaker_name, breaker_options})
   end
 
   @doc """
   Remove a circuit breaker.
   """
-  @spec remove(breaker_name :: term) :: ok_or_not_found
-  def remove(breaker_name) do
-    GenServer.call(__MODULE__, {:remove, breaker_name})
+  @spec remove(breaker_name :: term, process_name :: atom()) :: ok_or_not_found
+  def remove(breaker_name, process_name \\ __MODULE__) do
+    GenServer.call(process_name, {:remove, breaker_name})
   end
 
   @doc """
   Retrieve the configuration for a breaker.
   """
-  @spec get_config(breaker_name :: term) :: {:ok, BreakerConfiguration.t()} | {:error, :not_found}
-  def get_config(breaker_name) do
-    GenServer.call(__MODULE__, {:get_config, breaker_name})
+  @spec get_config(breaker_name :: term, process_name :: atom()) ::
+          {:ok, BreakerConfiguration.t()} | {:error, :not_found}
+  def get_config(breaker_name, process_name \\ __MODULE__) do
+    GenServer.call(process_name, {:get_config, breaker_name})
   end
 
   @doc """
   Retrieve a map with breaker names as keys and
   `BreakerBox.BreakerConfiguration` structs as values.
   """
-  @spec registered() :: %{optional(term) => BreakerConfiguration.t()}
-  def registered do
-    GenServer.call(__MODULE__, :registered)
+  @spec registered(process_name :: atom()) :: %{optional(term) => BreakerConfiguration.t()}
+  def registered(process_name \\ __MODULE__) do
+    GenServer.call(process_name, :registered)
   end
 
   @doc """
   Retrieve the status of a single breaker.
   """
-  @spec status(breaker_name :: term) :: fuse_status()
-  def status(breaker_name) do
-    GenServer.call(__MODULE__, {:status, breaker_name})
+  @spec status(breaker_name :: term, process_name :: atom()) :: fuse_status()
+  def status(breaker_name, process_name \\ __MODULE__)
+
+  def status(breaker_name, process_name) do
+    GenServer.call(process_name, {:status, breaker_name})
   end
 
   @doc """
   Retrieve the current status of all registered breakers.
   """
-  @spec status() :: %{optional(term) => fuse_status()}
-  def status do
-    GenServer.call(__MODULE__, :status)
+  @spec all_statuses(process_name :: atom()) :: %{optional(term) => fuse_status()}
+  def all_statuses(process_name \\ __MODULE__) do
```

Comment:
Because we added a default param, compiling fails with "conflicting defaults status/1 status/2" so we had to change this function name. Open to a better name.